File: ios/Simbi/Main/_SMBAnnotation.m
//
//  SMBAnnotation.m
//  Simbi
//
//  Created by flynn on 6/20/14.
//  Copyright (c) 2014 SimbiSocial. All rights reserved.
//

#import "_SMBAnnotation.h"


@interface _SMBAnnotation ()

@property (nonatomic, strong) SMBUser *user;

@end


@implementation _SMBAnnotation

- (instancetype)initWithUser:(SMBUser *)user
{
    self = [super init];
    
    if (self)
    {
        _user = user;
        _coordinate = CLLocationCoordinate2DMake(user.geoPoint.latitude, user.geoPoint.longitude);
    }
    
    return self;
}


- (MKAnnotationView *)annotationView
{
    MKAnnotationView *annotationView = [[MKAnnotationView alloc] initWithAnnotation:self reuseIdentifier:@"Maps"];
    
    [annotationView setEnabled:YES];
    [annotationView setCanShowCallout:NO];
    [annotationView setBackgroundColor:[UIColor simbiDarkGrayColor]];
    [annotationView.layer setBorderColor:[UIColor blackColor].CGColor];
    [annotationView.layer setBorderWidth:0.5f];
    [annotationView.layer setCornerRadius:22/2.f];
    
    UITapGestureRecognizer *tapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(selectAction:)];
    [annotationView addGestureRecognizer:tapGesture];
    
    [_user.profilePicture.thumbnailImage getDataInBackgroundWithBlock:^(NSData *data, NSError *error) {
        
        if (data)
        {
            UIImage *image = [[[UIImage imageWithData:data] imageWithSize:CGSizeMake(22, 22)] imageWithRoundedCornerSize:11.f];
            [annotationView setImage:image];
        }
    }];
    
    return annotationView;
}


- (void)selectAction:(id)sender
{
    if (_delegate)
        [_delegate annotation:self didSelectUser:_user];
}


@end

File: ios/Simbi/Main/SMBAnnotation.swift
//
//  SMBAnnotation.swift
//  Simbi
//
//  Created by flynn on 10/10/14.
//  Copyright (c) 2014 SimbiSocial. All rights reserved.
//

import MapKit
import UIKit


protocol SMBAnnotationDelegate: class {
    func annotationDidSelectUser(annotation: SMBAnnotation, user: SMBUser)
}


class SMBAnnotation: NSObject, MKAnnotation {
    
    weak var delegate: SMBAnnotationDelegate?
    
    let user: SMBUser
    var coordinate: CLLocationCoordinate2D
    override init() {
        self.user = SMBUser.currentUser()
        self.coordinate = CLLocationCoordinate2DMake(0, 0)
        super.init()
    }
    
    init(user: SMBUser) {
        
        self.user = user
        self.coordinate = CLLocationCoordinate2DMake(user.geoPoint.latitude, user.geoPoint.longitude)
        super.init()
    }

    
    func annotationView() -> MKAnnotationView {
        
        let annotationView = MKAnnotationView(annotation: self, reuseIdentifier: "Maps")
        
        annotationView.enabled = true
        annotationView.canShowCallout = false
        annotationView.backgroundColor = UIColor.simbiDarkGrayColor()
        annotationView.layer.borderColor = UIColor.simbiBlackColor().CGColor
        annotationView.layer.borderWidth = 0.5
        annotationView.layer.cornerRadius = 16
        annotationView.clipsToBounds = true
        
        let tapGesture = UITapGestureRecognizer(target: self, action: "selectAction:")
        annotationView.addGestureRecognizer(tapGesture)
        
//        user.profilePicture.thumbnailImage.getDataInBackgroundWithBlock { (data: NSData?, error: NSError?) -> Void in
//            
//            if data != nil {
//                let image: UIImage? = UIImage(data: data!)?.imageWithSize(CGSizeMake(22, 22)).imageWithRoundedCornerSize(11)
//                annotationView.image = image?
//            }
//        }
        
        return annotationView
    }
    
    
    func selectAction(sender: AnyObject) {
        delegate?.annotationDidSelectUser(self, user: user)
    }
}
